backends/ebpf/lower.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2017 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "lower.h"
+
+namespace EBPF {
+
+const IR::Expression* LowerExpressions::shift(const IR::Operation_Binary* expression) const {
+    auto rhs = expression->right;
+    auto rhstype = typeMap->getType(rhs, true);
+    if (rhstype->is<IR::Type_InfInt>()) {
+        auto cst = rhs->to<IR::Constant>();
+        mpz_class maxShift = Util::shift_left(1, LowerExpressions::maxShiftWidth);
```

Comment:
This seems like an expensive and obfuscated way of doing `mpz_class(1) << maxShiftWidth`

Reply:
I didn't realize that `<<` is defined for mpz_class.
The problem is really in the shift_left method, though.
I wonder whether it has the same semantics; the implementation is according to the spec.